include/range/v3/utility/associated_types.hpp

```diff
@@ -29,52 +29,41 @@ namespace ranges
         /// @{
 
         ////////////////////////////////////////////////////////////////////////////////////////
+        /// \cond
+        namespace detail
+        {
+            template<typename T, typename Enable = void>
+            struct difference_type2
+            {};
+
+            template<typename T>
+            struct difference_type2<T, meta::void_<decltype(std::declval<const T>() - std::declval<const T>())>>
```

Comment:
> `struct difference_type<T, meta::if_<`

I hope you mean `struct difference_type2`: I would hate to manually disambiguate this specialization with all of the `difference_type` specializations (and trigger https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78173 in the process).